### PR TITLE
test(iconprovider): add e2etest for iconprovider

### PR DESCRIPTION
**Pull request by Jason20020**
# Description
- Add E2E tests covering defaults, edge cases and allowed list testing
- Remove @tag in components 

## Links
https://jira-eng-gpk2.cisco.com/jira/browse/MOMENTUM-348

### CODE REVIEW

File: packages/components/config/playwright/setup/Components.page.ts [hunk 1]
```diff
@@ -147,6 +147,30 @@ class ComponentsPage {
...
+ * @param nestedLocator - An optional Playwright `Locator` representing a nested HTML element.
+ */
+  async setAttributes(locator: Locator, attributes: Record<string, string>, nestedLocator?: Locator) {
+    await locator.evaluate((element, attrs) => {
+      Object.keys(attrs).forEach((key) => {
+        element.setAttribute(key, attrs[key]);
+      });
+    }, attributes);
+
+    if (nestedLocator) {
```
Comment (cscheffauer) [inline packages/components/config/playwright/setup/Components.page.ts:165]: suggestion: that nestedLocator doesn't really make sense for the util - like i would like to see the nestedLocator just using the setAttributes in the test code. This extends the scope of the function unnecessarily.

File: packages/components/config/playwright/setup/Components.page.ts [hunk 2]
```diff
@@ -147,6 +147,30 @@ class ComponentsPage {
     );
   }
 
+  /**
```
Comment (cscheffauer) [inline packages/components/config/playwright/setup/Components.page.ts:150]: suggestion: the setAttribute function can be removed in favor of the new one

File: packages/components/src/components/iconprovider/iconprovider.e2e-test.ts [hunk 3]
```diff
@@ -8,45 +8,76 @@ type SetupOptions = {
...
+    await componentsPage.mount({
+      html: renderIconProvider(`
+      <mdc-iconprovider 
+        url="${restArgs.url}" 
+        id="nested" 
+        ${restArgs.fileExtension ? `file-extension="${restArgs.fileExtension}"` : ''}
+        ${restArgs.lengthUnit ? `length-unit="${restArgs.lengthUnit}"` : ''}
+        ${restArgs.size ? `size="${restArgs.size}"` : ''}
+      >
+        <mdc-subcomponent-icon icon-label-prefix="Nested IconProvider File Extension: "></mdc-subcomponent-icon>
```
Comment (cscheffauer) [inline packages/components/src/components/iconprovider/iconprovider.e2e-test.ts:40]: issue: there are no tests for this, like its getting rendered but i don't see a line in the e2e test which actually checks this one.

**Review by cscheffauer — CHANGES_REQUESTED** [review]

**Review by cscheffauer — APPROVED** [review]
